backend/server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/ServerApiHelper.java

```diff
@@ -135,13 +137,15 @@ public static String concat(String baseUrl, String relativePath) {
       (relativePath.startsWith("/") ? relativePath.substring(1) : relativePath);
   }
 
-  public static RuntimeException handleError(HttpClient.Response toBeClosed) {
+  public static RuntimeException handleError(HttpClient.Response toBeClosed, Runnable invalidTokenCallback) {
     try (var failedResponse = toBeClosed) {
       if (failedResponse.code() == HttpURLConnection.HTTP_UNAUTHORIZED) {
+        invalidTokenCallback.run();
```

Comment:
I feel propagating this callback in many classes is a bit odd. As discussed maybe we could rely on exceptions and have a "middle -man" catching and acting on them